### PR TITLE
Adds universal module support for noir_wasm

### PR DESCRIPTION
**Pull request by kobyhallx**
# Related issue(s)

relates to #590

# Description

Adds universal module support for noir_wasm

## Summary of changes

No breaking changes,
`crates/wasm/build-wasm` - adds logic to build two versions of module
`crates/wasm/wasm.nix` - NixOS helper to pin build packages and tools
`crates/fm/src/file_reader.rs` - replacing snippet with `@noir-lang/noir-source-resolver` module import

## Dependency additions / changes

N/A

## Test additions / changes

N/A

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

N/A

### CODE REVIEW

File: crates/wasm/wasm.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ nixpkgs ? import <nixpkgs> {} }:
+
```
Comment (kevaundray) [inline crates/wasm/wasm.nix:2]: what is the purpose of nix here? This module only requires rust so should build reliably I think

File: crates/wasm/build-wasm [hunk 2]
```diff
@@ -2,12 +2,24 @@
...
-cat $PKG_JSON | jq '.files += ["noir_wasm_bg.wasm.d.ts", "snippets/*"]' | tee $PKG_JSON
+wasm-pack build --scope noir-lang --target web --out-dir pkg/web
+
+COMMIT_SHORT=$(git rev-parse --short HEAD)
+VERSION_APPENDIX=""
+if [ -n "$COMMIT_SHORT" ]; then
+    VERSION_APPENDIX="-$COMMIT_SHORT"
+else
+    VERSION_APPENDIX="-NOGIT"
+fi
```
Comment (TomAFrench) [inline crates/wasm/build-wasm:19]: this is unused, seems like we want to use this in the `jq` lines.

File: crates/wasm/wasm.nix (hunk 1 above)
Comment (kobyhallx) [reply crates/wasm/wasm.nix:2]: nix file is left for convenience but is not mandatory for build

File: crates/wasm/build-wasm (hunk 2 above)
Comment (kobyhallx) [reply crates/wasm/build-wasm:19]: Thanks for spotting - fixed in next commit

**Review by kevaundray — APPROVED** [review]

**Pull request merged**